### ISSUE TITLE
Error messages are not perceivable by screen readers.

### ISSUE THREAD

**Issue by anirudhabhowmikhpe** [grommet/grommet#7877]
**Description:**
If errors occur when entering data in input fields, error messages appear in the affected field before the form is submitted. However, these are not output immediately or when the field is focused by the screen reader. Users cannot see that an error has occurred and submit the form. After submission, the errors are not displayed immediately, the keyboard focus remains on the "Save" button, and if the keyboard focus is moved to an incorrect field, the screen reader does not indicate that an error has occurred. Since the error messages are not displayed in the label of the affected field, blind users must find them by reading linearly. This user group therefore has difficulty perceiving errors and cannot correct them efficiently.

**Example:** Error messages are not perceivable in workspace details pages.

In addition, there are forms with multiple required fields. However, only the error message for the first affected field is displayed. Once the error has been corrected there, the error message for the next error appears. None of the error messages contain any suggestions for correction.

**Example:** Error messages are not displayed for all errors when adding a device.

**Proposed fix:**

Error messages should be output directly by the screen reader; to this end, they should be included in the label of the input field, for example.

In the event of an error, the focus should be on the first field affected after a form is submitted.

If multiple errors occur, each error should receive its own error message. All error messages
can be displayed collectively above the form, for example, with links to the individual errors.

These areas can be implemented as aria-live regions, for example, so that all error messages are output immediately by the screen reader, see
https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Live_Regions 

**Please take reference of this ticket:** https://hpe.atlassian.net/browse/GLCP-297839

**Comment by halocline** [comment]
Hi @anirudhabhowmikhpe - Thank you for this issue.

This behavior needs to be incorporated into your Form's `onValidate` callback. Please see how the [Shipping example](https://design-system.hpe.design/templates/forms#shipping) handles errors on submit. Note how focus is placed on the first form field with an error.

```js
  // On long forms, we want to focus the first of any fields that
  // return an error or info message. This removes the need for the
  // user to scroll to find which field blocked form submission.
  const onValidate = validationResults => {
    const target = formFields.find(
      field =>
        field in validationResults.errors || field in validationResults.infos,
    );
    if (target) {
      const targetFormField = document.getElementsByName(target);
      targetFormField[0].focus();
    }
  };
```

**Comment by jcfilben** [comment]
Marking as closed, see above response
